### PR TITLE
Implement ```create_card system```

### LINKED ISSUE #63: Implement create_card system
### Description

Each card must be created in the game(According to their stats) , we need a create_card system to create each card in order to make the player deck.

When a card is created it is assigned to each player’s deck in order to compose their team.

### PR DESCRIPTION
**Pull request by WhoIsNac**
```create_card``` implementation

Card creation / Stats assignment and card comparison working.

Fixes #63

### CODE REVIEW

File: contracts/src/tests/create_card.cairo [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    let world = spawn_world();
+
+    //Create Card For test prupose
+    let mut create_card_calldata: Array = ArrayTrait::new();
+    create_card_calldata.append(1); //Token_id
+    create_card_calldata.append(22); //Dribble
+    create_card_calldata.append(17); //Defense
+    create_card_calldata.append(10); //cost
+    create_card_calldata.append(3); //Role 0:Goalkeeper / 1:Defender etc..
+    create_card_calldata.append(0); //is_captain 0:false - 1:true
```
Comment (LucasLvy) [inline contracts/src/tests/create_card.cairo:27]: use the `array![]` macro

File: contracts/src/tests/create_card.cairo [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    world.execute('create_card_system', create_card_calldata);
+
+    let mut create_card_calldata_player2: Array<felt252> = ArrayTrait::new();
+    create_card_calldata_player2.append(2); //Token_id
+    create_card_calldata_player2.append(10); //Dribble
+    create_card_calldata_player2.append(15); //Defense
+    create_card_calldata_player2.append(10); //cost
+    create_card_calldata_player2.append(1); //Role 0:Goalkeeper / 1:Defender etc..
+    create_card_calldata_player2.append(0); //is_captain 0:false - 1:true
```
Comment (LucasLvy) [inline contracts/src/tests/create_card.cairo:37]: use the `array![]` macro

**Review by LucasLvy — CHANGES_REQUESTED** [review]

File: contracts/src/tests/create_card.cairo (hunk 1 above)
Comment (WhoIsNac) [reply contracts/src/tests/create_card.cairo]: array![] macro fix (https://github.com/keep-starknet-strange/tsubasa/pull/69/commits/7a163385b3d9d515ba23edd951b26e61aa85bf24)

File: contracts/src/tests/create_card.cairo (hunk 2 above)
Comment (WhoIsNac) [reply contracts/src/tests/create_card.cairo]: [array![] macro fix](https://github.com/keep-starknet-strange/tsubasa/pull/69/commits/1c75992736d8e544201ddb8fe967849184806b18)

File: contracts/src/components.cairo [hunk 3]
```diff
@@ -36,10 +37,22 @@ enum Roles {
...
+//     fn returnRole(self: Role) {
+//         match self {
+//             Roles::Goalkeeper => 'Goalkeeper'.print(),
+//             Roles::Defender => 'Defender'.print(),
+//             Roles::Midfielder => 'Midfielder'.print(),
+//             Roles::Attacker => 'Attacker'.print(),
+//         }
+//     }
+// }
+
```
Comment (LucasLvy) [inline contracts/src/components.cairo:55]:
```suggestion
```

File: contracts/src/components.cairo [hunk 4]
```diff
@@ -36,10 +37,22 @@ enum Roles {
 impl RolesSerdeLen of dojo::SerdeLen<Roles> {
     #[inline(always)]
     fn len() -> usize {
-        1
+        2
```
Comment (LucasLvy) [inline contracts/src/components.cairo:40]: revert this, the size is 1

File: contracts/src/components.cairo [hunk 5]
```diff
@@ -1,6 +1,7 @@
 use starknet::ContractAddress;
 use debug::PrintTrait;
 use option::{Option, OptionTrait};
+use core::traits::TryInto;
```
Comment (LucasLvy) [inline contracts/src/components.cairo:4]: Useless
```suggestion
```

File: contracts/src/tests/create_card.cairo [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    let world = spawn_world();
+
+    let mut create_card_calldata = array![];
+    create_card_calldata.append(1); //Token_id
+    create_card_calldata.append(22); //Dribble
+    create_card_calldata.append(17); //Defense
+    create_card_calldata.append(10); //cost
+    create_card_calldata.append(3); //Role 0:Goalkeeper / 1:Defender etc..
+    create_card_calldata.append(0); //is_captain 0:false - 1:true
```
Comment (LucasLvy) [inline contracts/src/tests/create_card.cairo:26]:
```suggestion
    let mut create_card_calldata = array![1, 22, 17, 10, 3, 0];
```

File: contracts/src/tests/create_card.cairo [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+    world.execute('create_card_system', create_card_calldata);
+
+    let mut create_card_calldata_player2 = array![];
+    create_card_calldata_player2.append(2); //Token_id
+    create_card_calldata_player2.append(10); //Dribble
+    create_card_calldata_player2.append(15); //Defense
+    create_card_calldata_player2.append(10); //cost
+    create_card_calldata_player2.append(1); //Role 0:Goalkeeper / 1:Defender etc..
+    create_card_calldata_player2.append(0); //is_captain 0:false - 1:true
```
Comment (LucasLvy) [inline contracts/src/tests/create_card.cairo:36]:
```suggestion
    let mut create_card_calldata_player2 = array![2, 10, 15, 10, 1, 0];
```

File: contracts/src/tests/create_card.cairo [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    create_card_calldata_player2.append(0); //is_captain 0:false - 1:true 
+
+    world.execute('create_card_system', create_card_calldata_player2);
+
+    let mut token_id_player1: u256 = 1;
+    let mut token_id_player2: u256 = 2;
+
+    let card_player1 = get!(world, token_id_player1, Card);
+    let card_player2 = get!(world, token_id_player2, Card);
+    //card_player1.role.print();
```
Comment (LucasLvy) [inline contracts/src/tests/create_card.cairo:45]:
```suggestion
```

**Review by LucasLvy — CHANGES_REQUESTED** [review]

File: contracts/src/tests/create_card.cairo (hunk 8 above)
Comment (WhoIsNac) [reply contracts/src/tests/create_card.cairo]: [review fix](https://github.com/keep-starknet-strange/tsubasa/pull/69/commits/a1ee4350e0b92ada5d02744875b6b08aee6adecb)

File: contracts/src/tests/create_card.cairo (hunk 7 above)
Comment (WhoIsNac) [reply contracts/src/tests/create_card.cairo]: [review fix](https://github.com/keep-starknet-strange/tsubasa/pull/69/commits/a1ee4350e0b92ada5d02744875b6b08aee6adecb)

File: contracts/src/tests/create_card.cairo (hunk 6 above)
Comment (WhoIsNac) [reply contracts/src/tests/create_card.cairo]: [review fix](https://github.com/keep-starknet-strange/tsubasa/pull/69/commits/a1ee4350e0b92ada5d02744875b6b08aee6adecb)

File: contracts/src/components.cairo (hunk 5 above)
Comment (WhoIsNac) [reply contracts/src/components.cairo]: [review fix](https://github.com/keep-starknet-strange/tsubasa/pull/69/commits/a1ee4350e0b92ada5d02744875b6b08aee6adecb)

File: contracts/src/components.cairo (hunk 4 above)
Comment (WhoIsNac) [reply contracts/src/components.cairo]: [review fix](https://github.com/keep-starknet-strange/tsubasa/pull/69/commits/a1ee4350e0b92ada5d02744875b6b08aee6adecb)

File: contracts/src/components.cairo [hunk 9]
```diff
@@ -40,6 +40,7 @@ impl RolesSerdeLen of dojo::SerdeLen<Roles> {
     }
 }
 
+
```
Comment (LucasLvy) [inline contracts/src/components.cairo:43]:
```suggestion
```

File: contracts/src/systems/create_card.cairo [hunk 10]
```diff
@@ -0,0 +1,30 @@
+#[system]
+mod create_card_system {
+    use traits::Into;
+    use dojo::world::Context;
+    use tsubasa::components::{Game,Roles, Card};
+
+
+    fn execute(
```
Comment (LucasLvy) [inline contracts/src/systems/create_card.cairo:8]: add documentation

File: contracts/src/tests/create_card.cairo [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+    world.execute('create_card_system', create_card_calldata_player2);
+
+    let mut token_id_player1: u256 = 1;
+    let mut token_id_player2: u256 = 2;
+
+    let card_player1 = get!(world, token_id_player1, Card);
+    let card_player2 = get!(world, token_id_player2, Card);
+
+    assert(card_player1.is_captain == false, 'is_captain is wrong');
+    assert(card_player1.defense == 17, 'defense is wrong');
```
Comment (LucasLvy) [inline contracts/src/tests/create_card.cairo:34]: test the full card ?

**Review by LucasLvy — CHANGES_REQUESTED** [review]

File: contracts/src/components.cairo (hunk 9 above)
Comment (WhoIsNac) [reply contracts/src/components.cairo]: [review fix#2](https://github.com/keep-starknet-strange/tsubasa/pull/69/commits/9bae1b4ce9f9e06baf851d41262c1400e70842a8)

File: contracts/src/systems/create_card.cairo (hunk 10 above)
Comment (WhoIsNac) [reply contracts/src/systems/create_card.cairo:15]: [review fix#2](https://github.com/keep-starknet-strange/tsubasa/pull/69/commits/9bae1b4ce9f9e06baf851d41262c1400e70842a8)

File: contracts/src/tests/create_card.cairo (hunk 11 above)
Comment (WhoIsNac) [reply contracts/src/tests/create_card.cairo]: [review fix#2](https://github.com/keep-starknet-strange/tsubasa/pull/69/commits/9bae1b4ce9f9e06baf851d41262c1400e70842a8) full card test added

File: contracts/src/components.cairo [hunk 12]
```diff
@@ -25,7 +25,7 @@ struct Card {
 }
 
 /// Available roles for cards
-#[derive(Copy, Drop, Serde)]
+#[derive(Copy,PartialEq, Drop, Serde)]
```
Comment (LucasLvy) [inline contracts/src/components.cairo:28]:
```suggestion
#[derive(Copy, PartialEq, Drop, Serde)]
```

File: contracts/src/systems/create_card.cairo [hunk 13]
```diff
@@ -0,0 +1,37 @@
+#[system]
+mod create_card_system {
+    use traits::Into;
+    use dojo::world::Context;
+    use tsubasa::components::{Game,Roles,Card};
+
+    /// Creates a card.
+    /// *`ctx` - Dojo context.
```
Comment (LucasLvy) [inline contracts/src/systems/create_card.cairo:8]:
```suggestion
    /// Creates a card.
    ///
    /// # Arguments
    ///
    /// *`ctx` - Dojo context.
```

File: contracts/src/systems/create_card.cairo [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+    /// * `token_id: The NFT token ID of the card being created.
+    /// * `dribble`: The dribble stat of the card being created.
+    /// *`defense`: The defense stat of the card being created.
+    /// *`cost`: The cost of the card being created.
+    /// *`role`: The role of the card being created.
+    /// *`is_captain`: Whether the card being created is the captain.
```
Comment (LucasLvy) [inline contracts/src/systems/create_card.cairo:14]:
```suggestion
    /// * `token_id - The NFT token ID of the card being created.
    /// * `dribble` - The dribble stat of the card being created.
    /// * `defense` - The defense stat of the card being created.
    /// * `cost` - The cost of the card being created.
    /// * `role` - The role of the card being created.
    /// * `is_captain` - Whether the card being created is the captain.
```

File: contracts/src/tests/create_card.cairo [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+
+#[test]
+#[available_gas(30000000)]
+fn test_create_card() {
+    let player1 = starknet::contract_address_const::<0x1>();
+    let player2 = starknet::contract_address_const::<0x2>();
+
+    let world = spawn_world();
+
+    //Token_id/Dribble/Defense/Cost/Role 0:Goalkeeper / 1:Defender etc../is_captain 0:false - 1:true
```
Comment (LucasLvy) [inline contracts/src/tests/create_card.cairo:20]:
```suggestion
    // Token_id, Dribble, Defense, Cost, Role
```

**Review by LucasLvy — CHANGES_REQUESTED** [review]

File: contracts/src/tests/create_card.cairo (hunk 15 above)
Comment (WhoIsNac) [reply contracts/src/tests/create_card.cairo]: [review fix#3](https://github.com/keep-starknet-strange/tsubasa/pull/69/commits/66d6985eafc3684db525216c9ddf650e5d87531e)

File: contracts/src/systems/create_card.cairo (hunk 14 above)
Comment (WhoIsNac) [reply contracts/src/systems/create_card.cairo]: [review fix#3](https://github.com/keep-starknet-strange/tsubasa/pull/69/commits/66d6985eafc3684db525216c9ddf650e5d87531e)

File: contracts/src/systems/create_card.cairo (hunk 13 above)
Comment (WhoIsNac) [reply contracts/src/systems/create_card.cairo]: [review fix#3](https://github.com/keep-starknet-strange/tsubasa/pull/69/commits/66d6985eafc3684db525216c9ddf650e5d87531e)

**Review by LucasLvy — DISMISSED** [review]

**Pull request merged**